collector/mdadm_linux.go

```diff
@@ -76,6 +77,26 @@ func evalStatusline(statusline string) (active, total, size int64, err error) {
 	return active, total, size, nil
 }
 
+func evalRaid0line(statusline string) (size int64, err error) {
+	matches := raid0lineRE.FindStringSubmatch(statusline)
+
+	// +1 to make it more obvious that the whole string containing the info is also returned as matches[0].
```

Comment:
This is part of the documented regexp package interface. It's also a very common interface in many other languages. This makes the code just harder to read and follow, I'd remove this.